lib/Parse/Lexer.cpp

```diff
@@ -91,18 +91,12 @@ static bool EncodeToUTF8(unsigned CharValue,
   return false;
 }
 
-
-/// CLO8 - Return the number of leading ones in the specified 8-bit value.
-static unsigned CLO8(unsigned char C) {
-  return llvm::countl_one(uint32_t(C) << 24);
-}
-
 /// isStartOfUTF8Character - Return true if this isn't a UTF8 continuation
 /// character, which will be of the form 0b10XXXXXX
 static bool isStartOfUTF8Character(unsigned char C) {
   // RFC 2279: The octet values FE and FF never appear.
   // RFC 3629: The octet values C0, C1, F5 to FF never appear.
-  return C <= 0x80 || (C >= 0xC2 && C < 0xF5);
+  return C < 0x80 || (C >= 0xC2 && C < 0xF5);
```

Comment:
Out of curiosity: Do you know of any character that uses a `0x80` continuation byte? If so, it would be nice to add a test case containing it.

Reply:
@ahoppen 
Added a test case that uses edge continuation bytes (0x80 and 0xBF), and another case for the lexer to accurately diagnose when there's a 0x80 in the source.